src/cloud-foundry-api-client.js

```diff
@@ -15,6 +19,29 @@ class CloudFoundryAPIClient {
       )).then(body => this._filterAppsResponse(JSON.parse(body)));
   }
 
+  getBuildContainersState() {
+    return this.fetchBuildContainers()
+      .then((buildContainers) => {
+        const numBuildContainers = buildContainers.length;
+        const startedContainers = buildContainers.filter(bc => bc.state === STATE_STARTED);
+        const allStarted = startedContainers.length === expectedNumBuildContainers;
+
+        if (numBuildContainers < expectedNumBuildContainers) {
+          return { error: `Expected ${expectedNumBuildContainers} build containers but only ${numBuildContainers} found.` };
+        }
+
+        if (!allStarted) {
+          return { error: `Not all build containers are in the ${STATE_STARTED} state.` };
```

Comment:
Any chance of a false positive here if we catch a build container that is in the middle of restarting?

Reply:
That's a good question. Maybe we should instead look for containers in the `STOPPED` state?

Reply:
Yeah, that works. I think we'll know pretty soon if we've made a mistake and can revisit this then.

Reply:
After thinking about it some more, I actually think we should keep it as looking for `STARTED`. We'll want to know if something gets stuck in `PENDING` or some other state. After monitoring the prod containers for a bit, I haven't noticed them changing their state from `STARTED` even while builds are being staged.

But, if I did miss something, having this running in staging for the next week should let us catch it and adjust.

Reply:
👍